doc: update migration guide and release notes for PRs #71827 and #72243 (#73508)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-3.7.rst

  of other Kconfig options which are now named `CONFIG_MBEDTLS_SHA*`.
* The `CONFIG_MBEDTLS_MAC_ALL_ENABLED` Kconfig option has been removed. Its equivalent is the
  combination of :kconfig:option:`CONFIG_MBEDTLS_HASH_ALL_ENABLED` and :kconfig:option:`CONFIG_MBEDTLS_CMAC`.
* When there is any PSA crypto provider available in the system
  (i.e. :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT` is set), desired PSA crypto

>   features must be explicitly enabled using proper `CONFIG_PSA_WANT_*`. (:github:`72243`)

Reviewer: ```suggestion
  features must be explicitly enabled using proper ``CONFIG_PSA_WANT_*``. (:github:`72243`)
```

Reviewer: Just realized that this change is to be done in other entries (mine) that are already merged. Can you do that for me (Mbed TLS entries in the migration guide)? :smiley:

Author: Done ;)

  of other Kconfig options which are now named `CONFIG_MBEDTLS_SHA*`.
* The `CONFIG_MBEDTLS_MAC_ALL_ENABLED` Kconfig option has been removed. Its equivalent is the
  combination of :kconfig:option:`CONFIG_MBEDTLS_HASH_ALL_ENABLED` and :kconfig:option:`CONFIG_MBEDTLS_CMAC`.
* Choice symbols :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_LEGACY_RNG` and

>   :kconfig:option:`MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` were added to allow the user

Reviewer 2: ```suggestion
  :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` were added to allow the user
```

  to specify how Mbed TLS PSA crypto core should generate random numbers. The former

>   option, which is the default, relies on legacy entropy and ctr/hmac DRBGs modules,

Reviewer: ```suggestion
  option, which is the default, relies on legacy entropy and CTR/HMAC DRBG modules,
```

>   while the latter relies on CSPRNG drivers. (:github:`71827`)

Reviewer: Wondering, is this entry relevant to the migration guide, or only to the release notes?

  to receive all the network packets. See details in
  https://www.man7.org/linux/man-pages/man7/packet.7.html documentation. (:github:`73338`)

* TCP now uses SHA-256 instead of MD5 for ISN generation. The crypto support for
  this hash computation was also changed from Mbed TLS to PSA APIs. This was achieved

>   by making :kconfig:option:`CONFIG_NET_TCP_ISN_RFC6528` depending on

Reviewer: ```suggestion
  by making :kconfig:option:`CONFIG_NET_TCP_ISN_RFC6528` depend on
```

  to receive all the network packets. See details in
  https://www.man7.org/linux/man-pages/man7/packet.7.html documentation. (:github:`73338`)

> * TCP now uses SHA256 instead of MD5 for ISN generation. The crypto support for

Reviewer: ```suggestion
* TCP now uses SHA-256 instead of MD5 for ISN generation. The crypto support for
```

> * TCP now uses SHA256 instead of MD5 for ISN generation. The crypto support for
>   for this hash computations was also changed from Mbed TLS to PSA APIs. (:github:`71827`)

Reviewer 2: ```suggestion
* TCP now uses SHA256 instead of MD5 for ISN generation. The crypto support for
  this hash computation was also changed from Mbed TLS to PSA APIs. (:github:`71827`)
```
